bugsnag/src/main/java/com/bugsnag/logback/BugsnagMarker.java

```diff
@@ -0,0 +1,79 @@
+package com.bugsnag.logback;
+
+import com.bugsnag.callbacks.Callback;
+
+import org.slf4j.Marker;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Marker used to pass a callback through a logger call
+ */
+public class BugsnagMarker implements Marker {
+
+    private static final String BUGSNAG_MARKER_NAME = "BUGSNAG_MARKER";
+
+    private Callback callback;
+
+    private List<Marker> references = new ArrayList<Marker>();
+
+    public BugsnagMarker(Callback callback) {
+        this.callback = callback;
+    }
+
+    public Callback getCallback() {
+        return callback;
+    }
+
+    public void setCallback(Callback callback) {
+        this.callback = callback;
+    }
+
+    @Override
+    public String getName() {
+        return BUGSNAG_MARKER_NAME;
+    }
+
+    @Override
+    public void add(Marker reference) {
+
```

Comment:
Shouldn't there be a `references.add` here?

Reply:
added tests